accepted/2024/release-staging-and-deprecation.md

```diff
@@ -0,0 +1,193 @@
+
+When staged, packages will not be visible / accessible to anyone but the co-owners. The co-owners will be able to see the current staged list for a given group using `nuget list --stage [group-id]` command. 
+
+
+#### Group Ownership
+There are few interesting scenarios related to package ownership:
+1. The user has staged a package to a group for later publishing, but the ownership has changed, and the user is no longer
+ an owner for a package in a group. At this point, publishing request should fail with a message that the group has been modified
+ from its original set, and the outcome of the request may be different from what the user was expecting, indicating the package
+ that was moved out of the group. If the user is aware of the problem and still wants to move forward with the change,
+ they will have to pass an additional `--force` parameter to the publish command: `nuget push --stage <group-id> --force`.
+2. The group to be published no longer has any packages in it. In this case the command should fail detailing the reason for the failure.
+3. This will be the same but for deprecation. Essentially, any bulk operation using groups, where the set of the packages in the group
+ have changed after staging, should result in an error with details about the change. The `--force` flag should again be used
+ to ignore the change and move forward with execution of the request (either publishing or deprecation). To summarize,
+ the fundamental principle is that a user cannot deprecate a package they don't own.
+
+#### Lifespan of a staged group
+If users keep staging packages but do nothing with them, NuGet will become an ever-growing package graveyard,
+ incurring higher and higher costs over time simply for storing these packages. To avoid this problem, stages should be time-limited.
+ That is, there should be a maximum lifespan for a stage. Any updates to the stage will reset its TTL date to some amount (let's say 30 days). If within the next 30 days no changes are made to the stage, NuGet should then remove the stage (all its packages basically).
+
+#### Expanding groups
+It should be possible for a group to expand over time. This is currently a real scenario, where .NET GA SDK ships a set of packages,
+ and then follow-up SDK updates ship updated packages which still belong to the same .NET 8 package group.
+ Here how this will play out with the tooling support described above:
+1. During GA release, the release team will stage the set of RTM packages for the release. (nuget stage --group-id "net8.0" --package-id <package-file-path>)
+2. Then on the day of the release, all the staged packages will be published using `nuget push --group "net8.0"` command.
+   At this point, the group with id `net8.0` will be empty.
+3. Later, as new builds are being prepared for a patch release, a new set of packages will be staged, to be later published to the same group.
```

Comment:
Unclear how that works with group lifespan described above.

Reply:
I think there are two similar concepts here: stage and group.
The `stage` is the subset of the packages which belong to a `group` that are not yet published (just staged). So they're on the server, but not available to customers yet. The user has to make a gesture to publish a stage, so that these packages will extend the group and become available to all customers.
The `group` however, is basically the pool which has both published and non-published / staged parts to it.

Having said this, lifespan information mentioned earlier is applicable to the stage really and packages which have already been published are not impacted by the lifespan.